lib/config.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+
+let _       = require('lodash'),
+    path    = require('path');
+
+class Config {
+
+    constructor(pkg, options, basePath) {
+
+        this.options = options;
+        this.basePath = basePath;
+        this.cacheControlHeader = process.env.CACHE_CONTROL || this.options.maxAge || 'max-age=60';
+        this.customHeaders = this.options.customHeaders || [];
+        this.description = this.options.description || pkg.description;
+        this.env = process.env.ENVIRONMENT || process.env.NODE_ENV || this.options.host || 'development';
+        this.envProd = this.options.envProd || 'production';
+        this.port = process.env.PORT || this.options.port || 3000;
+        this.host = process.env.HOST || this.options.host || '0.0.0.0';
```

Comment:
I'd argue this "should" be the way @codeclout originally had it: argument > environment > default

I think this is more natural, if a person wrote ...

```javascript
let server = hapi.instance({ port: 4321 });
```
... they'd expect the server to launch on 4321, not to be overridden by PORT=whatever in the environment they were using. 

Aside from my subjective assessment (which suspect Brian shares, since he instinctively wrote it that way) environment override could cause a problem if someone ever tried to launch two servers the same environment or to launch on a pragmatically determined port that was different than the PORT defined in environment variables. 

```javascript
process.env.PORT // 3001
let server = hapi.instance({ port: 4321 }); // nope, will be 3001
let server2 = hapi.instance({ port: 4322 }); // lol, also 3001
```

If we were trying to avoid breaking changes, we'd might want to keep it environment > argument for that reason, but other changes in this PR are already breaking.

Longer-term if y'all have some face time I would like to shop the config approach we used on AMP, I think it is one of the few things that I really got right there.